common/src/main/java/org/apache/drill/common/exceptions/UserException.java

```diff
@@ -50,7 +50,15 @@
    */
   public static Builder memoryError(final Throwable cause) {
     return UserException.resourceError(cause)
-      .message(MEMORY_ERROR_MSG);
+      .message(MEMORY_ERROR_MSG).addContext(cause.getMessage());
+  }
+
+  public static Builder memoryError(final String message) {
+    Builder builder =  UserException.resourceError();
+    if (!message.isEmpty()) {
```

Comment:
Oups, this method doesn't set the proper error message to the user exception, it's causing TestOutOfMemoryOutcome to fail.